util/blockrules.js

```diff
@@ -225,5 +228,42 @@ class BlockRules
   }
 }
 
-module.exports.BlockRules = BlockRules;
+class AdBlockRules extends BlockRules
```

Comment:
I wonder if it might make sense to just merge this to have a single BlockRules?
On one hand, this is cleaner, but on the other, having a single handleRequest() may remove ambiguity about what handles blocking first? I guess could go either way for now. Can also check how Playwright would handle multiple intercepted requests.